### ISSUE TITLE
Problems with GM_setvalue and GM_getValue since FF 53

### ISSUE THREAD

**Issue by we1hnachtsmann** [greasemonkey/greasemonkey#2537]
Hi,
since FF 53 I have some problems with GM_setValue and GM_getValue. I think it is GM_setValue who produces an error sometimes and the scripts do not work properly any more.

I've had this in several scripts on differents PCs (as the scripts are used from many people). Always the problem was solved by going back to FF 52.

Is this a known problem ?

**Comment by janekptacijarabaci** [comment]
See https://github.com/greasemonkey/greasemonkey/pull/2507 (and comments and links there).

**Comment by we1hnachtsmann** [comment]
That means: there will be no correction of this mistake ?

**Comment by janekptacijarabaci** [comment]
@we1hnachtsmann 
I don't know.

**Comment by nammmmm** [comment]
hello,
im having the same issue.
When this Error is thrown, it clearly kills all timers etc used by Grzeasemonkey and kills all other running script (FOR ALL WEBSITES!).
Its being a "long" time this issue is known, its a pretty critical one. So i hope devs will make a fix soon, because im at a point where one of my script is throwing it EVERY time. so it becomes unusable at all.

**Comment by we1hnachtsmann** [comment]
It's the same for me and it really makes GreaseMonkey unusable with FF or you have to deny all updates for FF which is critical regarding security. All my scripts are concerned and it's a pity that this bug has not yet been fixed when it is known for such a long time.

**Comment by we1hnachtsmann** [comment]
Sorry, closed it by accident :-(

**Comment by nammmmm** [comment]
sorry to invade your bug, but im pretty sure its related.
I am having another issue with GM_setValue.
tried
GM_setValue('thevar', true);
Randomly throws and error like 

> uncaught exception: unknown (can't convert to string)  

So i said ok.. ill ease GM.. change it to 
GM_setValue('thevar', 'true');

SAME error... cant be a string more than that... lol.

So really something wrong is happening with GM_set/getValue since this FF update.
its a critical error also, happening randomly , but still happening a lot! so it breaks the script.

**Comment by arantius** [comment]
This should be fixed in: https://addons.mozilla.org/firefox/downloads/file/722216/greasemonkey-3.12beta1-fx.xpi?src=devhub

Testing this beta version and reporting back here would be very useful!

**Comment by we1hnachtsmann** [comment]
Looks very good until now. At least the scripts are running again and I had no visible error

**Comment by we1hnachtsmann** [comment]
Update: I tested FF on a VM with a complete fresh installation. Everything seemed to be okay and run.
Now I imported the variables for one script from one FF to this new FF and now it doesn't work any more at all.
But the error seems to be different.

SessionHistory: [Exception... "Could not convert JavaScript argument arg 0 [nsISerializationHelper.serializeToString]"  nsresult: "0x80570009 (NS_ERROR_XPC_BAD_CONVERT_JS)"  location: "JS frame :: resource://gre/modules/sessionstore/Utils.jsm :: serializePrincipal :: line 125"  data: no]
String.toLowerCase is deprecated; use String.prototype.toLowerCase instead  mootools-1.2.4-core-nc.js:1697:42
SessionHistory: [Exception... "Could not convert JavaScript argument arg 0 [nsISerializationHelper.serializeToString]"  nsresult: "0x80570009 (NS_ERROR_XPC_BAD_CONVERT_JS)"  location: "JS frame :: resource://gre/modules/sessionstore/Utils.jsm :: serializePrincipal :: line 125"  data: no]
SessionHistory: [Exception... "Could not convert JavaScript argument arg 0 [nsISerializationHelper.serializeToString]"  nsresult: "0x80570009 (NS_ERROR_XPC_BAD_CONVERT_JS)"  location: "JS frame :: resource://gre/modules/sessionstore/Utils.jsm :: serializePrincipal :: line 125"  data: no]
SessionHistory: [Exception... "Could not convert JavaScript argument arg 0 [nsISerializationHelper.serializeToString]"  nsresult: "0x80570009 (NS_ERROR_XPC_BAD_CONVERT_JS)"  location: "JS frame :: resource://gre/modules/sessionstore/Utils.jsm :: serializePrincipal :: line 125"  data: no]
SessionHistory: [Exception... "Could not convert JavaScript argument arg 0 [nsISerializationHelper.serializeToString]"  nsresult: "0x80570009 (NS_ERROR_XPC_BAD_CONVERT_JS)"  location: "JS frame :: resource://gre/modules/sessionstore/Utils.jsm :: serializePrincipal :: line 125"  data: no]

Seems to repeat again and again.
